ligninkmc/kmc_functions.py

```diff
@@ -389,10 +387,10 @@ def update_events(state_dict, adj, last_event, event_dict, rate_vec, rate_dict,
 
             # Take any event_dict to be modified out of the set so that we can replace with the updated event_dict
             for event in events_to_be_modified:
-                ev_hash = hash(event)
-                if ev_hash in event_dict:
-                    del (event_dict[ev_hash])
-                    del (rate_vec[ev_hash])
+                if event in event_list:
+                    event_index = event_list.index(event)
```

Comment:
I also tended to like the security (i.e. felt less bug-prone) that I got out of storing and retrieving by event hash value, rather relying on lists to not become disordered at some point by removing the wrong index.

Reply:
At this point, it is far, far more important to have repeatable, consistent results that to worry about optimizing. Both skilled programmers agreed that this would be the last concern. Of course, the size of N will depend on how many monomers we include. I do not expect this to get up into the hundreds of thousands.

Anything that is not tested is bug-prone. 

It is clear that hashing is causing a problem, and it should be removed.

Please merged this request.